src/prefabs/monthPickerInput.tsx

```diff
@@ -0,0 +1,338 @@
+import * as React from 'react';
+import {
+  BeforeCreateArgs,
+  Icon,
+  prefab as makePrefab,
+  PrefabComponentOption,
+} from '@betty-blocks/component-sdk';
+import { DateTimePicker } from './structures/DateTimePicker';
+
+const beforeCreate = ({
+  close,
+  components: {
+    Content,
+    Field,
+    Footer,
+    Header,
+    FormField,
+    Toggle,
+    PropertySelector,
+    Label,
+    TextInput: Text,
+    CircleQuestion,
+    BBTooltip,
+  },
+  prefab: originalPrefab,
+  save,
+  helpers,
+}: BeforeCreateArgs) => {
+  const {
+    BettyPrefabs,
+    prepareInput,
+    useModelIdSelector,
+    useActionIdSelector,
+    usePrefabSelector,
+    usePropertyQuery,
+    setOption,
+    createUuid,
+    useModelQuery,
+    createBlacklist,
+    useModelRelationQuery,
+  } = helpers;
+
+  const [propertyPath, setProperty] = React.useState<any>('');
+  const [variableInput, setVariableInput] = React.useState(null);
+  const modelId = useModelIdSelector();
+  const actionId = useActionIdSelector();
+  const selectedPrefab = usePrefabSelector();
+  const [model, setModel] = React.useState<any>(null);
+  const [propertyBased, setPropertyBased] = React.useState(!!modelId);
+  const [prefabSaved, setPrefabSaved] = React.useState(false);
+
+  const [validationMessage, setValidationMessage] = React.useState('');
+
+  const modelRequest = useModelQuery({
+    variables: { id: modelId },
+    onCompleted: (result) => {
+      setModel(result.model);
+    },
+  });
+
+  const validate = () => {
+    if (modelRequest.loading) {
+      setValidationMessage(
+        'Model details are still loading, please try submitting again.',
+      );
+      return false;
+    }
+
+    return true;
+  };
+
+  let name: string | undefined;
+  let propertyKind;
+  let propertyModelId;
+  const componentId = createUuid();
+
+  function isProperty(path: string) {
```

Comment:
This is conflicting with line 79. 
You state that the `path`  can only be of type `string`, and then at line 79 you check if its `not` a string.
I think you want something like:
`isProperty(path: Property | Array<string> | string`